src/hyperlane-adapter/HyperlaneSenderAdapterV2.sol

```diff
@@ -162,16 +165,8 @@ contract HyperlaneSenderAdapter is ISingleMessageDispatcher, IBatchedMessageDisp
     );
 
     // try to make gas payment, ignore failures
-    // destination gasAmount is hardcoded to 500k similar to Wormhole implementation
-    // refundAddress is set from MMS caller state variable
     try
-      igp.payForGas{ value: msg.value }(
-        hyperlaneMsgId,
-        dstDomainId,
-        500000,
-        //address(this)
-        msg.sender
-      )
+      igp.payForGas{ value: msg.value }(hyperlaneMsgId, dstDomainId, gasAmount, msg.sender)
```

Comment:
Moved gas amount to storage as gas amount is now a state variable set in the constructor